include/moveit_middleware_benchmark/scenarios/scenario_perception_pipeline.hpp

```diff
@@ -0,0 +1,163 @@
+{
+public:
+  /** \brief Constructor for interface to send goal poses
+   *  \param [in] move_group_interface_ptr The move_group_interface for sending goal poses
+   */
+  ScenarioPerceptionPipeline(std::shared_ptr<MoveGroupInterface> move_group_interface_ptr);
+
+  /** \brief Given a \e pose_list, sends poses in pose_list to plan,
+   *  if the pipeline in \e sendTargetPose returns true,
+   *  then success_number is increased.
+   *  If the pipeline in \e sendTargetPose returns false,
+   *  failure_number is increased.
+   *  \param [in] test_case The pose list to benchmark
+   *  \return success_number and failure_number in tuple respectively
+   */
+  std::tuple<int, int> runTestCase(const nav_msgs::msg::Path& pose_list);
+
+  /** \brief Given a \e target_pose, sends this pose in order to plan via move_group_interface,
+   *  if the planning is successful,
+   *  planning message is sent to execute trajectory. Then it is returned true.
+   *  If the planning is failed, the execution is ignored and then it is returned false
+   *  \param [in] target_pose The pose to plan
+   *  \return if the planning successful, returns true, or false
+   */
+  bool sendTargetPose(const geometry_msgs::msg::Pose& target_pose);
+
+private:
+  /** move_group_interface_ptr to communicate with move_grouo_server */
+  std::shared_ptr<MoveGroupInterface> move_group_interface_ptr_;
+};
+
+class ScenarioPerceptionPipelineFixture : public benchmark::Fixture
+{
+protected:
+  /* ros node shared ptr*/
+  rclcpp::Node::SharedPtr node_;
+
+  /* move_group_interface_ptr for communicating with move_group_server */
+  std::shared_ptr<MoveGroupInterface> move_group_interface_ptr_;
+
+  /* selected test case index for benchmarking */
+  size_t selected_test_case_index_;
+
+  /** the list of test cases to be tested */
+  std::vector<nav_msgs::msg::Path> test_cases_ = {};
+
+public:
+  ScenarioPerceptionPipelineFixture();
+
+  /** \brief This method runs once each benchmark starts
+   *  \param [in] state
+   *  \return nothing is returned
+   */
+  void SetUp(::benchmark::State& /*state*/);
+
+  /** \brief This method runs as soon as each benchmark finishes
+   *  \param [in] state
+   *  \return nothing is returned
+   */
+  void TearDown(::benchmark::State& /*state*/);
+
+  /** \brief Creates test_cases by filling \e test_cases_
+   *  If the planning is failed, the execution is ignored and then it is returned false
+   *  \return nothing is returned
+   */
+  void createTestCases();
+
+  /** \brief Given a \e test_case_index, returns relevant test case.
+   *  \param [in] test_case_index The index of test_case
+   *  \return selected test case
+   */
+  nav_msgs::msg::Path selectTestCase(size_t test_case_index);
+
+  /** \brief Given a \e yaml_file_name, reads test cases from given yaml file
+   *  and adds this read test cases to \e test_cases_
+   *  \param [in] yaml_file_name the path of yaml file which includes test cases
+   *  \return nothing is returned
```

Comment:
You can drop the `\return` comment for void functions

Reply:
Fixed in https://github.com/CihatAltiparmak/moveit_middleware_benchmark/pull/5/commits/916a595a895dfc8febc655658a490f4051c8efa4 commit